web/src/stream_data.ts

```diff
@@ -1047,18 +1047,18 @@ export function create_sub_from_server_data(
         ...attrs,
     };
 
-    if (attrs.partial_subscribers !== undefined) {
-        peer_data.set_subscribers(sub.stream_id, attrs.partial_subscribers, false);
-    } else {
-        peer_data.set_subscribers(sub.stream_id, subscriber_user_ids ?? []);
-    }
-
     clean_up_description(sub);
 
     stream_info.set(sub.stream_id, sub);
     stream_ids_by_name.set(sub.name, sub.stream_id);
     sub_store.add_hydrated_sub(sub.stream_id, sub);
 
+    if (attrs.partial_subscribers !== undefined) {
+        peer_data.set_subscribers(sub.stream_id, attrs.partial_subscribers, false);
```

Comment:
My other comment probably makes this a moot point, but whenever you encounter code like `foo(..., ...., false)`, you should fix the bare boolean.  It's really hard to know what `false` means here.  I know that in this case you were just moving code and not actually introducing the code smell, but it's something to watch for.

Reply:
It's not totally clear to me why the code block was even moved in the first place, but again, I think it's a moot point.